### PR TITLE
renamed parameters and functions, edited comments

### PR DESCRIPTION
**Pull request by romanatarango**
### Scope & Purpose

The community part of https://github.com/arangodb/enterprise/pull/805.

- [ ] :hankey: Bugfix (requires CHANGELOG entry)
- [ ] :pizza: New feature (requires CHANGELOG entry, feature documentation and release notes)
- [ ] :fire: Performance improvement
- [x] :hammer: Refactoring/simplification
- [ ] :book: CHANGELOG entry made

#### Backports:

- [ ] Backport for 3.9: *(Please link PR)*
- [ ] Backport for 3.8: *(Please link PR)*
- [ ] Backport for 3.7: *(Please link PR)*

#### Related Information

*(Please reference tickets / specification / other PRs etc)*

- [ ] Docs PR: 
- [x] Enterprise PR: https://github.com/arangodb/enterprise/pull/800
- [ ] GitHub issue / Jira ticket number:
- [ ] Design document: 

### Testing & Verification

*(Please pick either of the following options)*

- [x] This change is a trivial rework / code cleanup without any test coverage.
- [ ] The behavior in this PR was *manually tested*
- [ ] This change is already covered by existing tests, such as *(please describe tests)*.
- [ ] This PR adds tests that were used to verify all changes:
  - [ ] Added new C++ **Unit tests**
  - [ ] Added new **integration tests** (e.g. in shell_server / shell_server_aql)
  - [ ] Added new **resilience tests** (only if the feature is impacted by failovers)
- [ ] There are tests in an external testing repository:
- [ ] I ensured this code runs with ASan / TSan or other static verification tools

Link to Jenkins PR run:

### Documentation

> All new features should be accompanied by corresponding documentation. 
> Bugs and features should furthermore be documented in the CHANGELOG so that
> support, end users, and other developers have a concise overview. 

- [ ] Added entry to *Release Notes* 
- [ ] Added a new section in the *Manual* 
- [ ] Added a new section in the *HTTP API* 
- [ ] Added *Swagger examples* for the HTTP API  
- [ ] Updated license information in *LICENSES-OTHER-COMPONENTS.md* for 3rd party libraries

### External contributors / CLA Note 

Please note that for legal reasons we require you to sign the [Contributor Agreement](https://www.arangodb.com/documents/cla.pdf)
before we can accept your pull requests.

### CODE REVIEW

**Comment by hkernbach** [comment]
image
@romanatarango fyi: After the refactor is done, we need to remove all Methods which are being unused then from our SmartEdge Classes.

File: arangod/VocBase/LogicalCollection.cpp [hunk 1]
```diff
@@ -403,13 +403,13 @@ ErrorCode LogicalCollection::getResponsibleShard(arangodb::velocypack::Slice sli
...
   if (isSatToSmartEdgeCollection() || isSmartToSatEdgeCollection()) {
-    return createSmartToSatKey(input);
+    return createKeyBetweenSmartAndSat(input);
   }
   return keyGenerator()->generate();
 }
 
 #ifndef USE_ENTERPRISE
-std::string LogicalCollection::createSmartToSatKey(VPackSlice) {
+std::string LogicalCollection::createKeyBetweenSmartAndSat(VPackSlice) const {
```
Comment (hkernbach) [inline arangod/VocBase/LogicalCollection.cpp:412]:
```suggestion
std::string LogicalCollection::createKeyBetweenSmartAndSat(VPackSlice) {
```

Comment (hkernbach) [reply arangod/VocBase/LogicalCollection.cpp:412]: using keyGenerator is not allowed to be const. As @mchacki told me once, that the key generator uses / increments ticks to be able to create unique keys.

File: arangod/VocBase/LogicalCollection.h [hunk 2]
```diff
@@ -365,7 +365,7 @@ class LogicalCollection : public LogicalDataSource {
    * Enterprise only method. See enterprise code for implementation
    * Community has a dummy stub.
    */
-  std::string createSmartToSatKey(arangodb::velocypack::Slice input);
+  std::string createKeyBetweenSmartAndSat(arangodb::velocypack::Slice edgeSlice) const;
```
Comment (hkernbach) [inline arangod/VocBase/LogicalCollection.h:368]:
```suggestion
  std::string createKeyBetweenSmartAndSat(arangodb::velocypack::Slice edgeSlice);
```

Comment (hkernbach) [reply arangod/VocBase/LogicalCollection.h:368]: using keyGenerator is not allowed to be const. As @mchacki told me once, that the key generator uses / increments ticks to be able to create unique keys.

**Review by hkernbach — CHANGES_REQUESTED** [review]
using keyGenerator is not allowed to be const. As @mchacki told me once, that the key generator uses / increments ticks to be able to create unique keys.

**Review by mpoeter — APPROVED** [review]

**Pull request closed without merging**